### PR TITLE
hikey: Add BL32 (OP-TEE) support

### PR DESCRIPTION
**Pull request by vchong**
Signed-off-by: Victor Chong <victor.chong@linaro.org>

### CODE REVIEW

**Comment by vchong** [comment]
Added commit to fix checkpatch warning.

File: plat/hisilicon/hikey/platform.mk [hunk 1]
```diff
@@ -12,11 +23,11 @@ COLD_BOOT_SINGLE_CPU		:=	1
 PROGRAMMABLE_RESET_ADDRESS	:=	1
 
 # Process flags
+$(eval $(call add_define,HIKEY_TSP_RAM_LOCATION_ID))
 $(eval $(call add_define,CONSOLE_BASE))
 $(eval $(call add_define,CRASH_CONSOLE_BASE))
 $(eval $(call add_define,PLAT_PL061_MAX_GPIOS))
 $(eval $(call add_define,PLAT_PARTITION_MAX_ENTRIES))
-$(eval $(call FIP_ADD_IMG,SCP_BL2,--scp-fw))
```
Comment (danh-arm) [inline plat/hisilicon/hikey/platform.mk]: Removing this line looks off-topic and should be mentioned in the commit message.

File: plat/hisilicon/hikey/platform.mk [hunk 2]
```diff
@@ -82,3 +93,9 @@ BL31_SOURCES		+=	plat/hisilicon/hikey/hisi_sip_svc.c			\
 				lib/pmf/pmf_smc.c
 endif
 
+# Enable workarounds for selected Cortex-A53 errata.
+ERRATA_A53_835769		:=	1
+ERRATA_A53_843419		:=	1
+ERRATA_A53_855873		:=	1
+ERRATA_A53_826319		:=	1
+ERRATA_A53_836870		:=	1
```
Comment (danh-arm) [inline plat/hisilicon/hikey/platform.mk]: Enabling these workarounds is off-topic and should be done in a separate commit. Some of them are also enabled in https://github.com/ARM-software/arm-trusted-firmware/pull/1013. Please agree with @hzhuang1 on which ones to enable.

File: plat/hisilicon/hikey/aarch64/hikey_common.c [hunk 3]
```diff
@@ -24,7 +24,7 @@
 					DEVICE_SIZE,			\
 					MT_DEVICE | MT_RW | MT_SECURE)
 
-#define MAP_TSP_MEM	MAP_REGION_FLAT(TSP_SEC_MEM_BASE, 		\
+#define MAP_TSP_MEM	MAP_REGION_FLAT(TSP_SEC_MEM_BASE,		\
```
Comment (danh-arm) [inline plat/hisilicon/hikey/aarch64/hikey_common.c]: The changes in this commit should be squashed with the previous commit

File: plat/hisilicon/hikey/platform.mk (hunk 1 above)
Comment (vchong) [inline plat/hisilicon/hikey/platform.mk]: This line is unnecessary and only generates a warning that it's overwriting some target in the main/root Makefile. Will add a description to the commit message.

File: plat/hisilicon/hikey/platform.mk (hunk 2 above)
Comment (vchong) [inline plat/hisilicon/hikey/platform.mk]: Ok, will remove these and let @hzhuang1 handle it in #1013.

File: plat/hisilicon/hikey/aarch64/hikey_common.c (hunk 3 above)
Comment (vchong) [inline plat/hisilicon/hikey/aarch64/hikey_common.c]: Ok. Will squash.

**Comment by vchong** [comment]
@hzhuang1 Can you please review? Thanks!

**Comment by vchong** [comment]
Comments addressed. Commits squashed and message updated. Thanks!

**Comment by danh-arm** [comment]
Sorry but following the change of all .md files to .rst, you will need to rebase this and update to use the reStructured text format.

**Comment by vchong** [comment]
That's fine. Had to do it anyway to address some of @hzhuang1 comments. : )

**Comment by vchong** [comment]
Replaced by https://github.com/ARM-software/arm-trusted-firmware/pull/1028.
